cadc-log/src/main/java/ca/nrc/cadc/log/WebServiceLogInfo.java

```diff
@@ -172,18 +174,17 @@ private void populateList(StringBuilder sb, List<?> o) {
         sb.append("[");
         if (!(o.isEmpty())) {
             boolean isFirstElement = true;
-            if (o.get(0) instanceof String) {
-                @SuppressWarnings("unchecked")
-                List<String> elements = (List<String>) o;
-                for (String element : elements) {
-                    if (isFirstElement) {
-                        isFirstElement = false;
-                    } else {
-                        sb.append(",");
-                    }
-                    sb.append("\"");
-                    sb.append(element);
-                    sb.append("\"");
+            for (Object element : o) {
+                String val = sanitize(element);
+                if (isFirstElement) {
+                    isFirstElement = false;
+                } else {
+                    sb.append(",");
+                }
+                if (element instanceof String || element instanceof UUID) {
```

Comment:
Looks much better.  I guess the only question left is whether to put double quotes around all list elements (not just String and UUID).  This is a question of whether `o` is a complex object or a primitive.  Since it's going through the sanitize() method already, I think we can only support non-complex elements.  So, what do you think of removing this check (line 184) and putting double quotes around all elements?

Reply:
If we do that, List<Integer> would produce incorrect JSON format, e.g. "intList":["1","2","3"]. With the simple-minded algorithm that we have currently implemented, we cannot expect to be able to cover all cases. We should expect developers who will add new attributes in the future to update the algorithm to produce the correct JSON for log info.

Reply:
I added UUID in the unit test to show that even for simple classes, toString() does not solve the problem for us. In the case of UUID, we cannot just enclose the sanitized value within [ and ] but need to add " and " instead.

Reply:
Okay, sounds good.